jena-fuseki2/jena-fuseki-geosparql/pom.xml

```diff
@@ -34,225 +34,206 @@
+              <goals>
+                <goal>jar-no-fork</goal>
+              </goals>
+            </execution>
+            <execution>
+              <id>attach-sources-test</id>
+              <goals>
+                <goal>test-jar-no-fork</goal>
+              </goals>
+            </execution>
+          </executions>
+
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-jar-plugin</artifactId>
+          <configuration>
+            <archive>
+              <manifestEntries>
+                <Automatic-Module-Name>${automatic.module.name}</Automatic-Module-Name>
+              </manifestEntries>
+            </archive>
+          </configuration>
+          <executions>
+            <execution>
+              <goals>
+                <goal>test-jar</goal>
+              </goals>
+            </execution>
+          </executions>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-javadoc-plugin</artifactId>
+          <configuration>
+            <source>8</source>
+            <version>true</version>
+            <show>public</show>
+            <quiet>true</quiet>
+            <encoding>UTF-8</encoding>
+            <windowtitle>Apache Jena GeoSPARQL</windowtitle>
+            <doctitle>Apache Jena GeoSPARQL ${project.version}</doctitle>
+            <bottom>Licenced under the Apache License, Version 2.0</bottom>
+          </configuration>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-shade-plugin</artifactId>
+          <configuration>
+            <shadedArtifactAttached>false</shadedArtifactAttached>
+            <transformers>
+              <!-- @@@ TODO @@ -->
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                <mainClass>org.apache.jena.fuseki.geosparql.Main</mainClass>
+              </transformer>
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheLicenseResourceTransformer" />
+              <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheNoticeResourceTransformer">
+                <addHeader>false</addHeader>
+              </transformer>
+            </transformers>
+            <filters>
+              <filter>
+                <artifact>*:*</artifact>
+                <excludes>
+                  <!-- Some jars are signed but shading breaks that.
+                       Don't include signing files.
+                  -->
+                  <exclude>META-INF/*.SF</exclude>
+                  <exclude>META-INF/*.DSA</exclude>
+                  <exclude>META-INF/*.RSA</exclude>
+                </excludes>
```

Comment:
Whew, I can only imagine all the troubleshoot and documentation-reading needed to get all these configuration settings right @afs :point_down: :point_up: Thanks a lot!

Reply:
There are certain patterns built-up over years, often by reading StackOverflow.

Does anyone ever write a POM file from scratch? 😃